### PR TITLE
Save current config to .yo-rc.json

### PR DESCRIPTION
**Pull request by pitrew**
This will save user answers to local storage as well. Then next time you execute yo it will first try to find global storage then local and override with local prompts.

This is very useful when I produce multiple projects (components) using the same generator. Then next time I want to update I wont be answering same questions again.

### CODE REVIEW

**Comment by SBoudrias** [comment]
Sorry for the delayed review, I think I'll have some time to go over it tonight.

**Comment by SBoudrias** [comment]
This look good to me, mind adding unit tests?

**Comment by eddiemonge** [comment]
ping @pitrew

**Comment by pitrew** [comment]
Hey, I'll add them

**Comment by pitrew** [comment]
Hi I've added tests and fixed raw list. In tests it was getting array of strings (and sepeartor) but the function was doing `_.map(ans, 'value')` whlist the input was a string. This was causing not saving defaults for lists when specified as arrays only

File: lib/util/prompt-suggestion.js [hunk 1]
```diff
@@ -54,15 +54,21 @@ var getListDefault = function (question, defaultValue) {
  *
  * @param  {Object}       question Inquirer prompt item
  * @param  {String|Array} answer   The inquirer answer
+ * @param  {Boolean}      storeAll Should store default values
```
Comment (SBoudrias) [inline lib/util/prompt-suggestion.js]: I wonder if there's really any case where we don't want to store the default value. Given that we only update the default value of the question array, it wouldn't ever be a problem to store the default one right?

File: test/prompt-suggestion.js [hunk 2]
```diff
@@ -293,6 +331,44 @@ describe('PromptSuggestion', function () {
...
+        default: 0,
+        store: true,
+        choices: ['foo', new inquirer.Separator('spacer'), 'bar', 'baz']
+      };
+
+      var mockAnswers = {
+        respuesta: 'foo'
+      };
+
+      this.store.delete('promptValues');
```
Comment (SBoudrias) [inline test/prompt-suggestion.js]: This should be done as part of the teardown.

File: lib/util/prompt-suggestion.js (hunk 1 above)
Comment (pitrew) [inline lib/util/prompt-suggestion.js]: @SBoudrias The original behaviour was not to store default values. I didn't want to break this. So storeAll set to false will reflect the original workflow. What do you suggest?

File: test/prompt-suggestion.js (hunk 2 above)
Comment (pitrew) [inline test/prompt-suggestion.js]: This is rather a setup. I moved it to beforeEach

**Comment by pitrew** [comment]
ping @eddiemonge, @SBoudrias

**Comment by SBoudrias** [comment]
Okay, sorry for the time it took me to review this. I had to sit down and wrap my head around the impact and make sure it wouldn't be problematic. So yeah, that all sounds good. Let's merge :)

Thanks for taking the time.

**Pull request merged**